src/lib/registration/widgets/package_search.rb

```diff
@@ -0,0 +1,215 @@
+          toggle_package
+        when "SelectionChanged"
+          update_details
+        end
+      end
+
+      # Determines whether the event should start the search
+      #
+      # @param event [Hash] UI event to check
+      # @return [Boolean]
+      def start_search_event?(event)
+        event["WidgetID"] == "search_form_button" ||
+          (event["WidgetID"] == "search_form_text" && event["EventReason"] == "Activated")
+      end
+
+      # Performs the search and updates the packages table
+      #
+      # @param text [String] Text to search for
+      def search_package(text)
+        @search = ::Registration::PackageSearch.new(text: text)
+        Yast::Popup.Feedback(_("Searching..."), _("Searching for packages with the given name")) do
+          selected_package_names = @selected_packages.map(&:name)
+          @search.packages.each do |pkg|
+            pkg.select! if selected_package_names.include?(pkg.name)
+          end
+        end
+        packages_table.change_items(@search.packages)
+        update_details
+      end
+
+      # Finds out the current package which is selected in the packages table
+      #
+      # @return [RemotePackage,nil]
+      def find_current_package
+        return unless search && packages_table.value
+        search.packages.find { |p| p.name == packages_table.value }
+      end
+
+      # Selects/unselects the current package for installation
+      #
+      # It does nothing if the package is already installed.
+      def toggle_package
+        package = find_current_package
+        return if package.nil? || package.installed?
+
+        if package.selected?
+          unselect_package(package)
+        else
+          select_package(package)
+        end
+
+        packages_table.update_item(package)
+        update_details
+      end
+
+      # Selects the current package for installation
+      #
+      # If required, it selects the addon for registration.
+      def select_package(package)
+        addon = package.addon
+        return unless addon.registered? || addon.selected? || enable_addon?(addon)
+
+        addon.selected unless addon.registered? || addon.selected?
+        package.select!
+        @selected_packages << package
+      end
+
+      # Unselects the current package for installation
+      def unselect_package(package)
+        package.unselect!
+        @selected_packages.delete(package)
+      end
+
+      # Updates the package details widget
+      def update_details
+        current_package = find_current_package
+        package_details.update(current_package) if current_package
+      end
+
+      # Asks the user to enable the addon
+      #
+      # It omits the question if the addon is already registered or selected for registration.
+      #
+      # @param addon [Addon] Addon to ask about
+      def enable_addon?(addon)
+        message = format(
+          _("'%{name}' module is not enabled for this system.\nDo you want to enable it?"),
```

Comment:
There are modules and extensions

```suggestion
          _("'%{name}' module/extension is not enabled for this system.\nDo you want to enable it?"),
```

would be probably more correct.